daffodil-io/src/main/scala/org/apache/daffodil/io/DirectOrBufferedDataOutputStream.scala

```diff
@@ -765,15 +818,30 @@ class DirectOrBufferedDataOutputStream private[io] (var splitFrom: DirectOrBuffe
       // no fragment bytes anywhere - just take the bytes
 
       val nBytes = (bufferNBits / 8).toInt
-      val nBytesPut = directDOS.putBytes(ba, 0, nBytes, finfo)
+      val nBytesPut = {
+        if (this.bufferingJOS.isFile)
+          directDOS.putFile(
```

Comment:
Is a flush here needed to make sure that the data is actually written to the file before we start reading it?